contracts/utils/test/lib_bytes.ts

```diff
@@ -1074,5 +1074,63 @@ describe('LibBytes', () => {
             expect(result).to.eq(byteArrayLongerThan32Bytes);
         });
     });
+
+    describe('writeLength', () => {
+        it('should return a null byte array if length is set to 0', async () => {
+            const result = await libBytes.publicWriteLength.callAsync(
+                byteArrayLongerThan32Bytes,
+                constants.ZERO_AMOUNT,
+                constants.NULL_BYTES,
+            );
+            expect(result).to.eq(constants.NULL_BYTES);
+        });
+        it('should return the same byte array if length is unchanged', async () => {
+            const byteLen = (byteArrayLongerThan32Bytes.length - 2) / 2;
+            const result = await libBytes.publicWriteLength.callAsync(
+                byteArrayLongerThan32Bytes,
+                new BigNumber(byteLen),
+                constants.NULL_BYTES,
+            );
+            expect(result).to.eq(byteArrayLongerThan32Bytes);
+        });
+        it('should shave off lower order bytes if new length is less than original', async () => {
+            const byteLen = (byteArrayLongerThan32Bytes.length - 2) / 2;
+            const newLen = new BigNumber(byteLen).dividedToIntegerBy(2);
+            const result = await libBytes.publicWriteLength.callAsync(
+                byteArrayLongerThan32Bytes,
+                newLen,
+                constants.NULL_BYTES,
+            );
+            expect(result).to.eq(
+                byteArrayLongerThan32Bytes.slice(
+                    0,
+                    newLen
```

Comment:
might be worth introducing a couple of const functions to convert from hex string length to byte length and vice versa

Reply:
I'm going to leave this for now. After switching to `fromHex(..).length`, I think this is the only place we are directly converting from byte length to length.